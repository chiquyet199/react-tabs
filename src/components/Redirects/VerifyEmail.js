import React from "react";
import queryString from "query-string";
import API from "../../utils/api";

export class VerifyEmailRedirect extends React.Component {
  componentWillMount = async () => {
    const {
      location: {search},
      go,
      storeUser,
    } = this.props;
    const {token} = queryString.parse(search);

    if (!token) {
      go("/verify-email");

      return;
    }

    const {success, user} = await API.verifyEmail(token).catch(() => ({
      success: false,
    }));

    if (!success) {
      // on fail or when user already verified
      // is not defined in requirements
      go("/verify-email");

      return;
    }

    storeUser(user);
    go("/verify-email-success");
  };

  render = () => null;
}
