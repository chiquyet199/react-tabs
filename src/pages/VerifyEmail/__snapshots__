/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/pages/VerifyEmail/__native_snapshots__