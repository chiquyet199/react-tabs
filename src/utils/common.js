import {Dimensions, Platform} from "react-native";
import React from "react";
import {connect} from "react-redux";
import styled from "styled-components/native";
import styledWeb, {withTheme} from "styled-components";

export const isWeb = Platform.OS === "web";
export const isNative = Platform.OS === "ios" || Platform.OS === "android";
export const isIos = Platform.OS === "ios";
export const isDev = process.env.NODE_ENV !== "production";
export const extractValueFromEvent = event =>
  event.target ? event.target.value : event;
export const extractKeyFromEvent = event =>
  event.nativeEvent ? event.nativeEvent.key : event.key;

export const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));

export const cleanProps = props => {
  // List of attributes to skip.
  const BLACKLISTED_PROPS = ["dispatch"];
  // Initialize the `cleanProps` as an empty object.
  const ownCleanProps = {};
  // Iterate over all the properties (Eslint - loops should be avoided in favor of array iterations)
  Object.keys(props).forEach(propKey => {
    // Check if the key is not in the blacklisted array.
    if (BLACKLISTED_PROPS.indexOf(propKey) === -1) {
      // Append the attribute to the clean object.
      ownCleanProps[propKey] = props[propKey];
    }
  });

  // Return the cleaned properties.
  return ownCleanProps;
};

export const responsive = Component => styles => {
  const ResponsiveComponent = styled(Component)`
    ${({size}) => styles[size]};
  `;

  const ResponsiveWrapper = props => {
    const ownProps = cleanProps(props);

    return <ResponsiveComponent {...ownProps} />;
  };

  return connect(state => ({
    size: state.dimensions.size,
  }))(ResponsiveWrapper);
};

export const responsiveWeb = Component => styles => {
  const ResponsiveWebComponent = styledWeb(Component)`
    ${({size}) => styles[size]};
  `;

  const ResponsiveWebWrapper = props => {
    const ownProps = cleanProps(props);

    return <ResponsiveWebComponent {...ownProps} />;
  };

  return connect(state => ({
    size: state.dimensions.size,
  }))(ResponsiveWebWrapper);
};

export const isX = (() => {
  const d = Dimensions.get("window");
  const {height, width} = d;

  return (
    // This has to be iOS duh
    Platform.OS === "ios" &&
    // Accounting for the height in either orientation
    (height === 812 || width === 812)
  );
})();

// Use theme fallback because Selva has another reducer
export const themed = Component =>
  connect(({globaltheme, theme}) => ({
    globaltheme: globaltheme || theme,
  }))(withTheme(Component));

export const redirectToBaseRoute = (
  goTo,
  form,
  location = {},
  prefixOnly = false,
) => {
  const {pathname} = location;
  const urlArr = pathname.split("/").filter(i => i !== "");
  const [baseUrl] = urlArr;
  if (prefixOnly) {
    return baseUrl;
  }
  const {DynamicForm} = form;
  if (!DynamicForm) {
    goTo(`/${baseUrl}`);
  }

  return true;
};

export const parseStyle = style => {
  return Object.keys(style).reduce((previousValue, currentValue) => {
    const starter = previousValue ? `${previousValue};\n` : "";

    return `${starter}${currentValue}: ${style[currentValue]}`;
  }, "");
};

export const applyStyleObjectToStyledComponent = (style, component) => {
  if (!style || Object.keys(style).length === 0) {
    return component;
  }

  const parsedStyles = parseStyle(style);

  return styled(component)`
    ${parsedStyles};
  `;
};
