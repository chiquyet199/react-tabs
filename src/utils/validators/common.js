export const isRequired = (value, key) => !value && {[key]: "required"};

export const validate = async (fields, values) => {
  const formValidations = mapValidators(fields, values);
  const results = await Promise.all(formValidations);

  return createErrorObject(results);
};

const mapValidators = (fields, values) =>
  Object.keys(fields).reduce((acc, key) => {
    const {validations = []} = fields[key];

    return [
      ...acc,
      ...validations.map(runValidation(values[key], key, values)),
    ];
  }, []);

const runValidation = (value, key, values) => async validator =>
  (await validator(value, key, values)) || {};

const createErrorObject = results =>
  results.reduce(
    (acc, err) => ({
      ...acc,
      ...err,
    }),
    {},
  ); // returns empty object if no errors

export const atLeastOneSelectedIsRequired = (value, key, values) => {
  const selectedValues = Object.values(values);
  const arrayIsEmpty = selectedValues.length === 0;
  const valueIsSelected = selectedValues.some(
    selectedValue => selectedValue === true,
  );

  if (arrayIsEmpty || !valueIsSelected) {
    return {
      [key]: "selectAtLeastOne",
    };
  }

  return true;
};
