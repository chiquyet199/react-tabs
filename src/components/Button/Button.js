import styled from "styled-components/native";
import Icon from "../Icon";
import {themed} from "../../utils/common";

export const BtnText = styled.Text`
  position: relative;
  top: -1px;
  font-size: ${({type, theme: {buttons}}) => buttons[type].fontSize || "16px"};
  font-weight: ${({type, theme: {buttons}}) =>
    buttons[type].fontWeight || "normal"};
  color: ${({type, theme: {buttons}}) => buttons[type].text};
`;

export const ButtonPart = styled.View`
  justify-content: center;
  flex: 1;
`;

export const Left = styled(ButtonPart)`
  position: absolute;
  height: 46px;
  width: ${({size}) => `${size}px`};
  left: ${({shift}) => (shift ? "10px" : 0)};
  top: 0;
  bottom: 0;
`;

export const Right = styled(ButtonPart)`
  position: absolute;
  height: 46px;
  width: ${({size}) => `${size}px`};
  right: ${({shift}) => (shift ? "10px" : 0)};
  top: 0;
  bottom: 0;
`;

export const Middle = styled(ButtonPart)`
  align-items: center;
  flex: 1;
  padding: 0;
  padding-right: ${({righticon, lefticon, size}) =>
    righticon || lefticon ? `${size + 10}px` : "10px"};
  padding-left: ${({lefticon, righticon, size}) =>
    lefticon || righticon ? `${size + 10}px` : "10px"};
  min-width: 60px;
`;

const Container = styled.TouchableOpacity`
  border-width: ${props => (props.noBorder ? 0 : 1)};
  width: 100%;
  max-width: 350px;
  margin: 0;
  padding: 0;
  height: 48px;
  border-radius: 6;
  border-color: ${({type, theme: {buttons}}) => buttons[type].border};
  background-color: ${({type, theme: {buttons}}) => buttons[type].background};
  opacity: ${({disabled}) => (disabled ? 0.5 : 1)};
`;

export const ButtonContainer = styled(Container)`
  padding: 0;
  margin: 5px;
  flex: auto;
  max-height: 48px;
  width: 100%;
  align-self: center;
`;

export const StyledIcon = styled(Icon)`
  text-align: center;
`;

export const IconContainer = styled(Container)`
  width: ${({size}) => size || 48}px;
  height: ${({size}) => size || 48}px;
  margin: 0;
  padding: 0;
  align-items: center;
  justify-content: center;
`;

export const MenuButtonIconContainer = styled(IconContainer)`
  margin-top: 0;
  margin-bottom: 2px;
  background-color: transparent;
`;

export const BounceContainer = styled.View`
  position: absolute;
  width: 200px;
  margin-top: ${({theme}) => theme.padding.top}px;
  margin-right: -20px;
  right: 0;
  z-index: 999;
`;

export const ButtonIcon = themed(styled(Icon)`
  display: flex;
  font-size: ${({size}) => `${size}px` || "48px"};
  margin-bottom: ${({bias}) => bias || "0"};
  color: ${({type, theme: {buttons}}) => buttons[type].text};
`);

export const MenuButtonText = styled.Text`
  font-size: 12px;
  color: ${({type, theme: {buttons}}) => buttons[type].text};
`;

export const TabContainer = styled(ButtonContainer)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  min-width: 142px;
  width: 100%;
  max-height: 38px;
  margin: 0;
  padding: 10px 30px;
  border-bottom: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
`;
export const TabText = styled(BtnText)`
  top: 0;
  color: inherit;
`;
