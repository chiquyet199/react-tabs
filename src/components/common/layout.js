import React from "react";
import styled from "styled-components/native";
import {isNative, isWeb, responsive, themed} from "../../utils/common";
import {TouchableOpacity} from "react-native";
import {DESKTOP_CONTAINER_WIDTH} from "../../constants/sizes";
import {ButtonContainer} from "../Button/Button";
import {func, node, number, string} from "prop-types";

export const FlexContainer = styled.View`
  flex: 1;
  width: 100%;
  height: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
`;

export const Flexed = styled.View``;

export const Flex = styled.View`
  margin-top: ${({theme}) => theme.padding.top + "px"};
  position: absolute;
  width: 100%;
  background-color: transparent;
`;

export const FlexColumn = styled.View`
  max-width: 537px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const TopBarContainer = styled.View`
  width: 100%;
  max-width: 100%;
  padding: 0;
  padding-top: ${({theme}) => theme.padding.top + "px"}
  z-index: 2;
  position: relative;
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].topbar.background}
`;

export const BaseContainer = styled.View`
  flex: 1;
  padding: 0;
  width: 100%;
  z-index: 1;
`;

export const CenteredContainer = styled.View`
  flex: 1;
  justify-content: ${({align}) => align || "center"};
  align-self: center;
  align-items: center;
  width: 100%;
  max-width: ${({full}) => (full ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`)};
  margin-horizontal: auto;
  padding-horizontal: ${({full}) => (full === "true" ? "0" : "10px")};
  flex-basis: auto;
`;

export const EndPageContainer = responsive(CenteredContainer)({
  S: `justify-content: flex-start`,
});

export const PageContainer = responsive(styled(CenteredContainer)`
  width: 100%;
  max-width: ${({full}) =>
    full === "true" ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  align-self: center;
  align-items: stretch;
  margin-top: ${({layout}) => (layout === "hq" ? "60px" : "40px")};
  ${({margintop}) => margintop >= 0 && `margin-top: ${margintop}px`};
  padding-horizontal: 10px;
  justify-content: center;
  flex-basis: auto;
`)({
  S: `
      padding-top: ${({theme}) => theme.padding.top + "px"};
      justify-content: space-between;
      padding-horizontal: 10px;
      margin-top: ${({layout}) => (layout === "hq" ? "80px" : "20px")};
    `,
});

export const FormContainer = responsive(styled.View`
  justify-content: center;
  align-items: center;
  max-width: ${DESKTOP_CONTAINER_WIDTH}px;
  width: 100%;
  align-self: center;
`)({
  S: `
      padding-top: 10px;
      justify-content: space-between;
      max-width: 100%;
    `,
});

export const FlexRow = styled.View`
  flex: 1;
  flex-direction: row;
  flex-basis: 100%;
`;

export const ButtonBar = styled.View`
  max-width: ${({isMenu}) =>
    isMenu ? "100%" : `${DESKTOP_CONTAINER_WIDTH}px`};
  width: 100%;
  border-top-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.lightSteel : theme.colors.none};
  border-top-width: ${({isMenu}) => (isMenu ? "1px" : 0)};
  background-color: ${({isMenu, theme}) =>
    isMenu ? theme.colors.white : theme.colors.none};
  flex-direction: row;
  margin-horizontal: 0;
  margin-bottom: ${({isMenu}) => (isMenu ? 0 : "5px")};
  flex-wrap: ${({wrap}) => (wrap ? "wrap" : "nowrap")};
  ${({mt}) => mt && `margin-top: ${mt}px}`};
  ${ButtonContainer} {
    margin: 20px;
  }
`;

export const BottomButtonBar = responsive(ButtonBar)({
  S: `flexGrow: 10;
      flexDirection: row;
      alignItems: center;
      justifyContent: flex-end;
    `,
});

export const ResponsiveWebScrollView = responsive(styled.ScrollView``)({
  M: `
        width: 500px;
  `,
  L: `
        width: 600px;
    `,
});

export const PageTitle = themed(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding-horizontal: 5px;
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  ${({mb}) => mb && `margin-bottom: ${mb}px}`};
  ${({color}) => color && `color: ${color}`};
`);

export const TextParagraph = themed(styled.Text`
  width: 100%;
  font-size: 16px;
  line-height: 24px;
  color: ${({disabled, theme, globaltheme}) =>
    disabled
      ? theme[globaltheme].disabledTextColor
      : theme[globaltheme].baseColor};
  font-weight: ${({bold}) => (bold ? "bold" : "normal")};
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  padding: ${({gutters}) => gutters || "15px 5px"};
  flex-shrink: 0;
  flex-grow: 0;
  ${({mb}) => mb && `margin-bottom: ${mb}px`};
  ${({color}) => color && `color: ${color}`};
`);

export const ViewParagraph = themed(styled.View`
  width: 100%;
  flex-shrink: 0;
  flex-grow: 0;
`);

export const SubTitle = themed(styled(TextParagraph)`
  font-weight: bold;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  ${({color}) => color && `color: ${color}`};
`);

export const TextLinkText = themed(styled.Text`
  text-align: center;
  font-size: 16;
  line-height: 24;
  color: ${({theme, globaltheme}) => theme[globaltheme].accentColor};
  font-weight: 700;
  padding-top: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")};
  padding-bottom: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-left: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  padding-right: ${({gutters}) => (gutters >= 0 ? `${gutters}px` : "15px")}
  text-align:  ${({align}) => align || "center"};
`);

export const TextLink = ({onClick, children, gutters, align, globaltheme}) => (
  <TouchableOpacity onPress={onClick}>
    <TextLinkText align={align} gutters={gutters} globaltheme={globaltheme}>
      {children}
    </TextLinkText>
  </TouchableOpacity>
);

TextLink.propTypes = {
  onClick: func,
  children: node,
  gutters: number,
  align: string,
  globaltheme: string,
};

export const NotchSafety = styled.View`
  height: 40px;
`;

export const PageHeader = styled.Text`
  font-size: 18;
  text-align: center;
  width: 100%;
  margin-top: ${({theme}) => theme.padding.top + 15 + "px"};
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  position: absolute;
`;

export const HeadlineText = responsive(styled.Text`
  width: 100%;
  font-size: 28px;
  line-height: 36px;
  color: ${({globaltheme, theme}) => theme[globaltheme].baseColor};
  align-self: ${({left}) => (left ? "flex-start" : "center")};
  text-align: ${({left}) => (left ? "left" : "center")};
  margin: 8px 0 40px;
  padding: 0;
`)({
  S: `
    padding-left: 15px;
    `,
  L: `
    align-self: center;
   `,
});

export const BoxButtonBar = styled.View`
  align-self: center;
  align-items: stretch;
  width: 100%;
  margin-horizontal: auto;
  flex-direction: row;
`;

export const FixedWidthContainer = styled.View`
  width: ${DESKTOP_CONTAINER_WIDTH}px;
  margin-horizontal: auto;
  padding-horizontal: 15px;
`;

export const PaddedContent = styled.View`
  padding-horizontal: 10px;
`;

export const ErrorMessage = themed(styled(TextParagraph)`
  color: ${({theme}) => theme.colors.alertRed};
`);

// Scroll layout

export const ScrollContentContainer = styled.View`
  width: 100%;
  ${isNative ? "flex: 1;" : "height: calc(100vh - 52px)"};
  flex-grow: ${() => (isWeb ? 0 : 1)};
  flex-shrink: ${() => (isWeb ? 0 : 1)};
  justify-content: center;
  align-self: center;
  align-items: center;
`;

export const ScrollButtonBar = styled(ButtonBar)`
  padding-horizontal: 10px;
  ${({mt}) => mt && `margin-top: ${mt}px}`};
`;

export const ScrollScrollView = styled.ScrollView`
  ${isNative ? "" : "height: 100%"};
  width: 100%;
  flex-grow: 1;
  flex-shrink: 1;
`;
