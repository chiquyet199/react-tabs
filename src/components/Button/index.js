import React from "react";
import {View} from "react-native";
import {
  array,
  bool,
  func,
  object,
  oneOf,
  oneOfType,
  number,
  string,
} from "prop-types";
import {
  ButtonContainer,
  Left,
  Middle,
  Right,
  BtnText,
  StyledIcon,
  IconContainer,
  MenuButtonIconContainer,
  BounceContainer,
  MenuButtonText,
  ButtonIcon,
  TabContainer,
  TabText,
} from "./Button";
import Icon from "../Icon";
import {applyStyleObjectToStyledComponent, themed} from "../../utils/common";

export const Button = ({
  left,
  right,
  middle,
  onClick,
  type,
  noborder,
  style,
  iconsize,
  theme,
  disabled,
}) => {
  const color = theme && theme.buttons[type].text;

  return (
    <ButtonContainer
      noborder={noborder}
      type={type}
      onPress={onClick}
      activeOpacity={0.6}
      style={style}
      theme={theme}
      disabled={disabled}
    >
      {left && (
        <Left size={iconsize || 32} shift={middle}>
          <StyledIcon
            type={type}
            name={left}
            size={iconsize || 32}
            color={color}
          />
        </Left>
      )}
      {middle && (
        <Middle lefticon={left} righticon={right} size={iconsize || 32}>
          <BtnText ellipsizeMode="tail" type={type} theme={theme}>
            {middle}
          </BtnText>
        </Middle>
      )}
      {right && (
        <Right size={iconsize || 32} shift={middle}>
          <StyledIcon
            type={type}
            name={right}
            size={iconsize || 32}
            color={color}
          />
        </Right>
      )}
    </ButtonContainer>
  );
};

Button.propTypes = {
  left: oneOfType([string, bool]),
  middle: string,
  right: oneOfType([string, bool]),
  onClick: func,
  type: string,
};

Button.defaultProps = {
  left: null,
  right: null,
  type: "primary",
};

export const IconButton = ({
  onClick,
  name,
  type,
  style,
  iconsize,
  bias,
  theme,
}) => (
  <IconContainer style={style} type={type} onPress={onClick}>
    <Icon
      size={iconsize}
      type={type}
      name={name}
      bias={bias}
      color={theme && theme.buttons[type].text}
    />
  </IconContainer>
);

IconButton.propTypes = {
  name: string.isRequired,
  onClick: func,
  type: string.isRequired,
};

const ThemedIconContainer = ({
  name,
  type,
  style,
  iconsize,
  bias,
  theme: {buttons},
}) => (
  <IconContainer size={iconsize} style={style} type={type}>
    <ButtonIcon
      size={iconsize}
      type={type}
      name={name}
      bias={bias}
      color={buttons[type].text}
    />
  </IconContainer>
);

ThemedIconContainer.propTypes = {
  name: string.isRequired,
  type: string.isRequired,
  style: oneOf([array, object]),
  iconsize: number,
  bias: number,
  theme: object.isRequired,
};

export const StyledIconContainer = themed(ThemedIconContainer);

export const MenuButton = ({
  onClick,
  name,
  type,
  iconContainerStyle,
  children,
  iconsize,
  theme,
}) => {
  const StyledInnerContainer = applyStyleObjectToStyledComponent(
    iconContainerStyle,
    MenuButtonIconContainer,
  );

  return (
    <View style={{flex: 1, alignItems: "center"}}>
      <StyledInnerContainer type={type} noBorder onPress={onClick}>
        <Icon
          type={type}
          name={name}
          size={iconsize || 32}
          color={theme.buttons[type].text}
        />
      </StyledInnerContainer>
      <MenuButtonText type={type}>{children}</MenuButtonText>
    </View>
  );
};

MenuButton.propTypes = {
  onClick: func,
  name: string,
  type: string,
  iconContainerStyle: object,
  iconsize: number,
  theme: object,
};

export const FormBounceButton = ({children, onClick, globaltheme, theme}) => (
  <BounceContainer>
    <Button
      onClick={onClick}
      middle={children}
      type={theme[globaltheme].form.bounceButton}
      noBorder
    />
  </BounceContainer>
);

FormBounceButton.propTypes = {
  children: string.isRequired,
  globaltheme: oneOf(["dark", "light"]),
  onClick: func.isRequired,
};

export const TabButton = ({onClick, left, iconsize, middle, type}) => (
  <TabContainer type={type} onPress={onClick}>
    <StyledIcon type={type} name={left} size={iconsize || 20} color="inherit" />
    <TabText ellipsizeMode="tail" type={type}>
      {middle}
    </TabText>
  </TabContainer>
);

TabButton.propTypes = {
  left: string,
  type: string.isRequired,
  middle: string.isRequired,
  iconsize: number,
  onClick: func,
};

export default themed(Button);
