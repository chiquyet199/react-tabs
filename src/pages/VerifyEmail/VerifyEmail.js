import React, {Component} from "react";
import {AsyncStorage} from "react-native";
import {themed} from "../../utils/common";
import Button from "../../components/Button";
import verifyEmailImage from "../../resources/images/verify-email.png";
import {
  CenteredContainer,
  PageTitle,
  TextParagraph,
  SubTitle,
} from "../../components/common/layout";
import {CenterIconImage} from "../../components/common/images";
import {shape, func, string} from "prop-types";

export class VerifyEmail extends Component {
  async componentDidMount() {
    await AsyncStorage.getItem("preActivationData");
    const {
      showTopbarCenterText,
      translations: {header},
    } = this.props;
    showTopbarCenterText(header);
  }

  componentWillMount() {
    const {email, usersId, checkPreVerificationLogin} = this.props;
    if (!email || !usersId) {
      checkPreVerificationLogin();
    }
  }

  componentWillUnmount() {
    const {hideTopbarCenterText} = this.props;
    hideTopbarCenterText();
  }

  render() {
    const {translations, requestVerifyEmail, email, go, userId} = this.props;

    return (
      <CenteredContainer>
        <CenterIconImage source={verifyEmailImage} />
        <PageTitle>{translations.verify__email__title__text}</PageTitle>
        <SubTitle>{email}</SubTitle>
        <TextParagraph>{translations.verify__email__text}</TextParagraph>
        <TextParagraph>{translations.not__received__an__email}</TextParagraph>
        <Button
          type="secondary"
          middle={translations.resend__verification__email}
          onClick={() => requestVerifyEmail({userId, email})}
        />
        <Button type="primary" middle="Home" onClick={() => go("/welcome")} />
      </CenteredContainer>
    );
  }
}

VerifyEmail.propTypes = {
  translations: shape({
    resend__verification__email: string.isRequired,
    verify__email__text: string.isRequired,
    verify__email__title__text: string.isRequired,
    not__received__an__email: string.isRequired,
    header: string.isRequired,
  }).isRequired,
  go: func.isRequired,
  email: string.isRequired,
  userId: string.isRequired,
  requestVerifyEmail: func.isRequired,
  showTopbarCenterText: func.isRequired,
  hideTopbarCenterText: func.isRequired,
  checkPreVerificationLogin: func.isRequired,
};

export default themed(VerifyEmail);
