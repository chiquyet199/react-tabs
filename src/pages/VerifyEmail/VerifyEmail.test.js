import React from "react";
import {shallow} from "enzyme";
import {SubTitle} from "../../components/common/layout";
import Button from "../../components/Button";
import {Platform} from "react-native";
import {VerifyEmail} from "./VerifyEmail";

import {extend} from "underscore";

describe("VerifyEmail", () => {
  it(`matches snapshot on ${Platform.OS}`, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  it("it displays the user's email", () => {
    const component = getComponent();
    expect(component.find(SubTitle).props().children).toEqual("max@gmail.com");
  });

  it("calls to checkPreVerificationLogin", () => {
    const checkPreVerificationLogin = jest.fn();
    getComponent({checkPreVerificationLogin});
    expect(checkPreVerificationLogin).toHaveBeenCalled();
  });

  it("calls requestVerifyEmail() on button click", () => {
    const requestVerifyEmail = jest.fn();
    const component = getComponent({requestVerifyEmail});
    component
      .find(Button)
      .first()
      .simulate("click");
    expect(requestVerifyEmail).toHaveBeenCalledWith({
      userId: "12345",
      email: "max@gmail.com",
    });
  });

  it("goes to welcome page when 'Home' buttons is clicked", () => {
    const go = jest.fn();
    const component = getComponent({go});
    component
      .find(Button)
      .find({type: "primary"})
      .simulate("click");
    expect(go).toHaveBeenCalledWith("/welcome");
  });

  const getMockProps = () => ({
    hideTopbarCenterText: () => {},
    showTopbarCenterText: () => {},
    requestVerifyEmail: () => {},
    checkPreVerificationLogin: () => {},
    go: () => {},
    userId: "12345",
    email: "max@gmail.com",
    theme: {},
    globaltheme: "light",
    isNative: false,
    translations: {
      header: "Verify email",
      verify__email__title__text: "Verify your email address",
      verify__email__text:
        "We need to know you are you before creating a " +
        "RentPassport. Please click the link in the email we sent you.",
      resend__verification__email: "Resend verification email",
      not__received__an__email: "Not received an email?",
    },
  });
  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<VerifyEmail {...parsedProps} />);
  };
});
