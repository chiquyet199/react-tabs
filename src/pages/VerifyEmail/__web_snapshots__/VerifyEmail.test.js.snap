// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyEmail matches snapshot on web 1`] = `
<Styled(View)>
  <Styled(Image)
    source="verify-email.png"
  />
  <Connect(WithTheme(Styled(Text)))>
    Verify your email address
  </Connect(WithTheme(Styled(Text)))>
  <Connect(WithTheme(Styled(Connect(WithTheme(Styled(Text))))))>
    max@gmail.com
  </Connect(WithTheme(Styled(Connect(WithTheme(Styled(Text))))))>
  <Connect(WithTheme(Styled(Text)))>
    We need to know you are you before creating a RentPassport. Please click the link in the email we sent you.
  </Connect(WithTheme(Styled(Text)))>
  <Connect(WithTheme(Styled(Text)))>
    Not received an email?
  </Connect(WithTheme(Styled(Text)))>
  <Connect(WithTheme(Button))
    middle="Resend verification email"
    onClick={[Function]}
    type="secondary"
  />
  <Connect(WithTheme(Button))
    middle="Home"
    onClick={[Function]}
    type="primary"
  />
</Styled(View)>
`;
